middleware/head_get.go

```diff
@@ -0,0 +1,23 @@
+package middleware
+
+import (
+	"net/http"
+
+	"github.com/go-chi/chi"
+)
+
+func HeadGet(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if r.Method == "HEAD" {
+			rctx := chi.RouteContext(r.Context())
+			h := rctx.Routes.FindHandler(chi.NewRouteContext(), "HEAD", rctx.RoutePath)
```

Comment:
I see you pass `chi.NewRouteContext()` as `rctx` into the `FindHandler()` method.

---

Do we ever need to pass real Context into it? And why?

I'm thinking if we could remove the rctx parameter from this method signature and pass the Context down to `tree.FindRoute()` from the `FindHandler()` method itself.

ie.
```go
h, _ := rctx.FindHandler("HEAD", rctx.RoutePath)

// the FindHandler() method could pass the rctx itself
// or chi.NewRouteContext()
// down to the mx.tree.FindRoute() method
```

Reply:
@VojtechVitek I agree that would be nice, except the `FindHandler` method under the hood is calling `mx.tree.FindRoute(rctx, m, path)` which is the call to search the routing tree for the matching handler. The tree routing method requires a *Context (route context) where it will store the url params it find during traversal. It's tuned carefully for low gc pressure. In the case that the url params are throw-away (like in this case), then we could add a chi.NoContext, make an interface, and allow tree.FindRoute to skip setting url params during traversal. I dont really want to do this right now. Rather, people can make a pool of route context's in their middleware or whereever instead right now.

I just realized a different problem though, which is that the `FindHandler` method does not traverse the subrouter trees, therefore the current implementation is actually incomplete. I will push a breaking test soon.